apps/files/src/components/FileSharingSidebar.vue

```diff
@@ -125,15 +104,23 @@ export default {
     selectedFile () {
       return this.highlightedFile
     },
-    $_ocCollaborators_users () {
-      return this.shares.filter(collaborator => {
-        return collaborator.info.share_type === '0' || collaborator.info.share_type === '6'
-      })
-    },
-    $_ocCollaborators_groups () {
-      return this.shares.filter(collaborator => {
-        return collaborator.info.share_type === '1'
-      })
+    $_ocCollaborators () {
+      return this.shares
+        .filter(collaborator => this.$_ocCollaborators_isUser(collaborator) || this.$_ocCollaborators_isGroup(collaborator))
+        .sort((c1, c2) => {
+          const name1 = c1.displayName.toLowerCase().trim()
+          const name2 = c2.displayName.toLowerCase().trim()
+          // sorting priority 1: display name (lower case, ascending), 2: share type (groups first), 3: id (ascending)
+          if (name1 === name2) {
+            if (this.$_ocCollaborators_isGroup(c1) === this.$_ocCollaborators_isGroup(c2)) {
+              return parseInt(c1.info.id, 10) < parseInt(c2.info.id, 10) ? -1 : 1
+            } else {
+              return this.$_ocCollaborators_isGroup(c1) ? -1 : 1
+            }
+          } else {
+            return name1 < name2 ? -1 : 1
```

Comment:
let's use natural sort, see https://github.com/owncloud/phoenix/blob/master/apps/files/src/fileSortFunctions.js#L28, it is used also in the file list

Reply:
Using natural sort on the string comparison now (last bit of the sorting). Moved natural sort into `helpers/textUtils.js` for that.